src/server/server.ts

```diff
@@ -210,6 +212,40 @@ registerRoute('post', '/compile/snippet', compileCommon);
  */
 registerRoute('post', '/compile/page', (req, res) => compileCommon(req, res, true /*wrapWithRunnerChrome*/));
 
+registerRoute('get', '/open-in-playground/:host/:type/:id/:filename', async (req, res) => {
+    let relativePath;
+    switch (req.params.host.toUpperCase()) {
+        case 'EXCEL':
+            relativePath = 'xl';
+            break;
+        case 'WORD':
+            relativePath = 'word';
+            break;
+        case 'POWERPOINT':
+            relativePath = 'ppt';
+            break;
+        default:
+            return;
+    }
+
+    return Promise.resolve(getVersionNumber()).then(versionNumber => {
+        let extractDirName = path.resolve(__dirname, `test${(new Date()).getTime()}`);
+        let zip = fs.createReadStream(path.resolve(__dirname, 'Doc1.zip')).pipe(unzip.Extract({ path: extractDirName }));
+        zip.on('close', () => {
+            let xmlFileName = `${extractDirName}/${relativePath}/webextensions/webextension1.xml`;
+            let xmlStringData = fs.readFileSync(xmlFileName, 'utf-8').toString();
+            xmlStringData = xmlStringData.replace('%placeholder_version%', versionNumber).replace('%placeholder_type%', req.params.type).replace('%placeholder_id%', req.params.id);
+            fs.writeFileSync(xmlFileName, xmlStringData);
+
+            res.attachment(req.params.filename);
+            const archiver = Archiver('zip');
+            archiver.pipe(res);
+            archiver.directory(extractDirName, '');
+            archiver.finalize();
+        });
```

Comment:
I think the "registerroute" is already adding a catch and will display an error page if something goes wrong.  That's part of the reasons that it makes sense for it to be a normal URL, in a way, so that on error we can show the error as a plain html page.